rllib/algorithms/sac/sac_catalog.py

```diff
@@ -0,0 +1,174 @@
+import gymnasium as gym
+import numpy as np
+
+# TODO (simon): Store this function somewhere more central as many
+# algorithms will use it.
+from ray.rllib.algorithms.ppo.ppo_catalog import _check_if_diag_gaussian
+from ray.rllib.core.models.catalog import Catalog
+from ray.rllib.core.models.configs import (
+    FreeLogStdMLPHeadConfig,
+    MLPEncoderConfig,
+    MLPHeadConfig,
+)
+from ray.rllib.core.models.base import Model, Encoder
+from ray.rllib.models.torch.torch_distributions import TorchSquashedGaussian
+from ray.rllib.utils.annotations import override, OverrideToImplementCustomLogic
+
+
+# TODO (simon): Check, if we can directly derive from DQNCatalog.
+# This should work as we need a qf and qf_target.
+# TODO (simon): Add CNNEnocders for Image observations.
+class SACCatalog(Catalog):
+    """The catalog class used to build models for SAC."""
+
+    def __init__(
+        self,
+        observation_space: gym.Space,
+        action_space: gym.Space,
+        model_config_dict: dict,
+    ):
+        """Initializes the SACCatalog.
+
+        Args:
+            observation_space: The observation space of the Encoder.
+            action_space: The action space for the Pi Head.
+            model_config_dict: The model config to use.
+        """
+
+        super().__init__(
+            observation_space=observation_space,
+            action_space=action_space,
+            model_config_dict=model_config_dict,
+        )
+
+        # Define the heads.
+        self.pi_and_qf_head_hiddens = self._model_config_dict["post_fcnet_hiddens"]
+        self.pi_and_qf_head_activation = self._model_config_dict[
+            "post_fcnet_activation"
+        ]
+
+        # We don't have the exact (framework specific) action dist class yet and thus
```

Comment:
Yeah, same in PPO. I feel like we should change the Catalog API here:
Provide the `framework` already in the c'tor (imo, there is no good reason NOT to do this), then users can already construct the pi-config in the c'tor, which is much cleaner.

Reply:
WOuld this also work with dynamic action spaces?

Reply:
But do we already support dynamic action spaces?

Reply:
The thing is that right now, we construct one module config in the c'tor and the other one during `build()`, which is pretty bad imo. We should be consistent in where and when we do what. :)

Reply:
No, we do not support them right now - at least not for the new stack I guess.